nntrainer/tensor/manager.cpp

```diff
@@ -212,11 +212,13 @@ Manager::AllocFunc Manager::getAllocFunc(bool is_weight) {
   } else if (!is_weight) {
     /** only for gradients */
     if (max_grad_size > 0 && enable_gradient_memory_opt) {
-      std::shared_ptr<float> window(new float[max_grad_size],
-                                    std::default_delete<float[]>());
+      std::shared_ptr<Tensor> shared_grad =
+        std::make_shared<Tensor>(TensorDim({(unsigned int)max_grad_size}));
```

Comment:
How about having a memory block instead of a tensor to add extra overhead?
We can add bufsize argument to Tensor::Map.

If you have an ongoing updates regarding this, I think it's fine.

Reply:
In an upcoming PR, this memory is going to be allocated later, when needed.
To allow lazy allocation, wrapping it with tensor will be required.

Reply:
I'll update tensor map for the future use anyway.
Could you update to `nntrainer::Tensor shared_grad` ? as it is captures it does not need to be a shared pointer or we can use unique_ptr

Reply:
Done. 
Updated tensor map as part of this PR itself.